File: utils/marathon-compose/marathon-compose.js
#!/usr/bin/env node

fs = require("fs");
yaml = require("js-yaml");

// FIXME: All of the following should be parameters
composeFilename = "../../docker-compose.yml"
groupId = "azure"

try {
    var compose = yaml.safeLoad(fs.readFileSync(composeFilename, 'utf8'));
    console.log("Supplied Docker Compose file:");
    console.log(JSON.stringify(compose, null, "    "));
} catch (e) {
    console.log(e);
}

var marathon = {};
marathon.id = groupId;
marathon.groups = [];
apps = [];
for (var id in compose) {
  app = {};
  app.id = id;
  
  app.container = {};
  app.container.type = "docker";
  app.container.docker = {};
  // FIXME: what if compose has a build rather than image?
  app.container.docker.image = compose[id].image;
  // FIXME: what can we get from the Docker Compose file for Network?
  app.container.docker.network = "HOST"  
  
  app.ports = compose[id].ports;
  //FIXME: this is likely too simplistic, what should we do
  app.requirePorts = true;  

  //FIXME: Dependencies need to be massaged
  app.dependencies = []
  for (var link in compose[id].links) {
    app.dependencies.push(compose[id].links[link]);
  }
  
  app.cpu = 0.5;
  app.mem = 500;
  app.instances = 1;
  apps.push(app); 
}
marathon.groups.push(apps)
console.log("Generated Marathon file:");
console.log(JSON.stringify(marathon, null, "    "))

File: utils/marathon-compose/marathon-compose.py
import json
import yaml

compose= """
  lb:
    image: eeacms/haproxy
    ports:
      - "80:80"
      - "1936:1936"
    links:
      - web
  web:
    build: web
    ports:
      - "80"
    links:
      - rest
  rest:
    build: java
    ports:
      - "8080:8080"
  load_test:
    build: load_test
    links:
      - web
"""
print "Compose in:"
print yaml.dump(yaml.load(compose))

desired_marathon = """
{
  "id": "azure",
  "groups": [
    {
      "apps": [
        {
          "id": "web",
          "cpus": 0.5,
          "mem": 500,
          "instances": 1,
          "ports": [
            80
          ],
          "requirePorts": true,
          "container": {
            "docker": {
             "image": "azuredemo/web:azurecon",
             "network": "HOST"
            },
            "type": "DOCKER"
          },
          "dependencies": [
            "/azure/demo/rest"
          ]
        },
        {
          "id": "rest",
          "cpus": 1.0,
          "mem": 500,
          "instances": 1,
          "ports": [
            8080
          ],
          "requirePorts": true,
          "container": {
            "docker": {
              "image": "azuredemo/rest:azurecon",
              "network": "HOST"
            },
            "type": "DOCKER"
          }
        },
        {
          "id": "lb",
          "cpus": 0.5,
          "mem": 500,
          "instances": 1,
          "ports": [
            80
          ],
          "env": {
            "BACKENDS": "web-demo-azure.marathon.mesos:80"
          },
          "requirePorts": true,
          "container": {
            "docker": {
              "image": "eeacms/haproxy",
              "network": "HOST"
            },
            "type": "DOCKER"
          },
          "dependencies": [
            "/azure/demo/web"
          ]
        }
      ],
      "id": "demo"
    }
  ]
}
"""

print "Desired Marathon file"
print json.dumps(desired_marathon, indent=4, separators=(',', ': '))

# docker-compose does not provide a group id
# FIXME: group ID should be provided in the command line
group="azure"

